### PR TITLE
Vanilla fixes for public pages

### LINKED ISSUE #2131: [Vanilla] Public snapcraft pages (IoT, publisher pages etc.)
### TODO:
- [x] ALL: navigation hover text color is too dark
- [x] HOME PAGE: clicking on first snap flow language links doesn't scroll panel into view properly
- [x] IOT: cards should use `p-card--highlighted` style
- [x] SEARCH: use `row` instead of `u-fixed-width`
- [ ] DISTRO: there is no space between last paragraph and heading in install instructions card (https://github.com/canonical-web-and-design/vanilla-framework/issues/2447)
- [x] DISTRO: featured snaps are misaligned

### PR DESCRIPTION
**Pull request by bartaz**
Fixes #2131 

### QA

Go to and review pages for any functionality and visual issues.

Demo: https://snapcraft-io-canonical-web-and-design-pr-2143.run.demo.haus/

This includes:
- navigation - make sure hover styles are good
- home page - make sure clicking on language selector for first snap flow works correctly
- IoT
- store and category pages
- search results

### CODE REVIEW

**Comment by Lukewh** [comment]
Clicking the fsf languages jumps back to the top of the page for me on Firefox?

**Comment by Lukewh** [comment]
On IoT page we can probably have:
`col-3 col-medium-3` for the snap cols so that it drops down to 2 columns on medium breakpoint?

File: static/sass/_snapcraft_p-featured-snap.scss [hunk 1]
```diff
@@ -32,7 +32,7 @@
     border-radius: $border-radius;
     column-gap: $sp-xxx-large;
     display: grid;
-    grid-template-columns: minmax(5rem, 43rem) minmax(1rem, 160px);
+    grid-template-columns: minmax(5rem, 50rem) minmax(1rem, 160px);
```
Comment (Lukewh) [inline static/sass/_snapcraft_p-featured-snap.scss]: good catch :+1:

File: static/sass/_snapcraft_p-featured-snap.scss [hunk 2]
```diff
@@ -57,8 +57,8 @@
     }
 
     .p-featured-snap__icon {
-      width: 9rem;
       height: 9rem;
+      width: 9rem;
```
Comment (Lukewh) [inline static/sass/_snapcraft_p-featured-snap.scss]: :pray: thanks you - one less warning

**Comment by therealjuan** [comment]
image

What is this vanilla peopleeee???

**Comment by bartaz** [comment]
@Lukewh @therealjuan Sorry, forgot one commit. This should fix the issues with first snap flow on home page

**Comment by therealjuan** [comment]
image

Maybe we need to look at the height of this element? It feels like as the grid is bigger the element maintains the height. Probably, might need some adjustment.

**Comment by kwm14** [comment]
## Design review
**Navigation**
- [ ] Hover state is good but we've removed the selected state keyline in Vanilla, selected state is now the same style as the hover state. Seen on https://jaas.ai/store and https://maas.io/install it would be great to be consistent with other applications 😄 
Screenshot 2019-08-06 at 13 41 30

**Homepage**
- [ ] Same as @Lukewh comment above. Clicking the FSF languages jumps back to the top of the page for me on Chrome
- [ ] As we're using `p-card--header` it's gives us the dividing lines, which looks nice. But once a language is clicked we get another line pointer. So I'd suggest just using `p-card`
Screenshot 2019-08-06 at 13 45 31

**Store**
- Looking good 👍 

**Store & category pages**
- Looking good 👍 

**Search results**
- Looking good 👍

**Comment by bartaz** [comment]
@therealjuan Yes, the page is wider now. So to keep banner aspect ratio we would need to make them higher as well. 
We might need to have a look at this as well when dealing with banner images.

**Comment by kwm14** [comment]
> **Homepage**
> 
> * [ ]  As we're using `p-card--header` it's gives us the dividing lines, which looks nice. But once a language is clicked we get another line pointer. So I'd suggest just using `p-card`
> 

Just seen you've fixed this in your last commit before I posted. So you can ignore @bartaz 😉

**Comment by bartaz** [comment]
@kwm14 Fixed the issues:

Languages on home page shouldn't have a border inside them now.

And selected style for nav is updated.
Screenshot 2019-08-06 at 15 26 10
But honestly speaking I'm not sure why in new nav we are making active/selected items less visible than the others... This gray on dark background looks disabled.

**Comment by therealjuan** [comment]
image

Just checking the account icon needs some margin?

**Comment by bartaz** [comment]
> Just checking the account icon needs some margin?

@therealjuan Ah, that's where we have this icon. I just forgot about it, seen some styles for it and was like "I don't think we need it anymore. DELETE". LOL.

**Comment by kwm14** [comment]
> Languages on home page shouldn't have a border inside them now.
> 
> And selected style for nav is updated.
> Screenshot 2019-08-06 at 15 26 10

@bartaz updates look good. Thanks 👍 

> But honestly speaking I'm not sure why in new nav we are making active/selected items less visible than the others... This gray on dark background looks disabled.

At first, it is different but you'll get used to it 😉 Alternatively, if we want to override in Snapcraft that's fine by me, keeping as is will be consistent with our other sites and applications 😄 

At least it's better than https://www.docker.com/products who have no selected state at all... 😆

**Comment by therealjuan** [comment]
For me it seems more logical that hovering a navigation link underlines text as with the rest of links on the site without changing the colour, at the end of the day these are text links. Then we can have the `#c9c9c9` colour for selected...

image

WDYGT?

**Comment by bartaz** [comment]
@therealjuan @kwm14 I'm merging this as it is (with Vanilla 2.0 navigation styles).

If we want to change the navigation let's have it as separate post-migration issue.

**Pull request merged**